app/javascript/src/Settings/components/Common/RadioFieldset.jsx

```diff
@@ -0,0 +1,50 @@
+// @flow
+
+import * as React from 'react'
+import { useState } from 'react'
+import { FormFieldset, FormLegend } from 'Form'
+import { Radio } from '@patternfly/react-core'
+import type { FieldGroupProps, FieldCatalogProps } from 'Settings/types'
+
+type CheckEvent = SyntheticEvent<HTMLButtonElement>
+
+type Props = FieldGroupProps & FieldCatalogProps
+
+const useSelectedOnChange = (value, onChange) => (
+  typeof onChange === 'function'
+    ? [value, onChange]
+    : useState(value).map(x => typeof x === 'function' ? (_c, e: CheckEvent) => x(e.currentTarget.value) : x)
```

Comment:
I don't think it is needed to do these kind of _setState_ transformations since `_c === e.currentTarget.value`. This PF's way to avoid us digging into events.